syne_tune/optimizer/schedulers/transfer_learning/__init__.py

```diff
@@ -46,13 +69,15 @@ def __init__(
             config_space: Dict,
             transfer_learning_evaluations: Dict[str, TransferLearningTaskEvaluations],
             metric_names: List[str],
+            **kwargs
     ):
         """
         A mixin that adds basic functionality for using offline evaluations.
         :param config_space: configuration space to be sampled from
         :param transfer_learning_evaluations: dictionary from task name to offline evaluations.
         :param metric_names: name of the metric to be optimized.
         """
+        super().__init__(config_space=config_space, **kwargs)
```

Comment:
What is this calling? There is no superclass.

Reply:
It will call the __init__ of any other extended class. In case of RUSHScheduler, this is the line that calls __init__ of HyperbandScheduler.

Reply:
Maybe you can explain this to me in our next meeting. I am coming from C++, where this multiple inheritance is always quite complicated. I think I need this mixin technique as well.